cloudweatherreport/utils.py

```diff
@@ -16,13 +15,59 @@
 import traceback
 import uuid
 import yaml
+import requests
 
-from jujuclient.juju1.environment import Environment as env1
-from jujuclient.juju2.environment import Environment as env2
+import jujuclient.juju1.environment
+import jujuclient.juju2.environment
+import jujuclient.juju1.facades
+import jujuclient.juju2.facades
 
 
 PROVISIONING_ERROR_CODE = 240
 
+ISO_TIME_FORMAT = "%Y-%m-%dT%H:%M:%S"
+
+
+def get_bundle_yaml(status):
+    if not status:
+        return None
+    if status.bundle_yaml:
+        return status.bundle_yaml
+    elif status.charm:
+        return create_bundle_yaml(status.charm.get("name"))
+    return None
+
+
+def serializer(obj):
+    """
+    Fall-back serializer.
+
+    Currently only handles datetime objects.
+    """
+    if isinstance(obj, datetime):
+        return obj.strftime(ISO_TIME_FORMAT)
+    else:
+        raise TypeError('%s is not serializable' % obj)
+
+
+def fetch_svg(bundle_yaml):
+    if not bundle_yaml:
+        return None
+    try:
+        r = requests.post('http://svg.juju.solutions', bundle_yaml)
+    except Exception:
+        logging.warn("Timeout exception from svg.juju.solution "
+                     "for \nbundle.yaml:\n{}".format(bundle_yaml),)
```

Comment:
I was trying to find those words when initially reviewing this block of code and you beat me to it. :)